### PR TITLE
feat: top reader notification

### PR DESCRIPTION
**Pull request by omBratteng**
## Changes

Also https://github.com/dailydotdev/daily-api/pull/2375

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://as-667-top-reader-notification.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/notifications/NotificationItemAvatar.tsx [hunk 1]
```diff
@@ -7,6 +7,9 @@ import SourceButton from '../cards/common/SourceButton';
 import { ProfileTooltip } from '../profile/ProfileTooltip';
 import { ProfileImageLink } from '../profile/ProfileImageLink';
 import { ProfileImageSize } from '../ProfilePicture';
+import { MedalBadgeIcon } from '../icons/MedalBadge';
```
Comment (rebelchris) [inline packages/shared/src/components/notifications/NotificationItemAvatar.tsx:10]: I think for icons we said we would keep using barrel imports as the only exception for now :)

File: packages/shared/src/components/modals/BootPopups.tsx [hunk 2]
```diff
@@ -248,28 +248,9 @@ export const BootPopups = (): ReactElement => {
     addBootPopup({
       type: LazyModal.TopReaderBadge,
       props: {
-        onAfterClose: (keywordValue: string) => {
+        onAfterClose: () => {
```
Comment (rebelchris) [inline packages/shared/src/components/modals/BootPopups.tsx:251]: Oh it's now like a extra afterClose?

**Review by rebelchris — APPROVED** [review]
Looks good to me :)

File: packages/shared/src/components/notifications/NotificationItemAvatar.tsx (hunk 1 above)
Comment (omBratteng) [reply packages/shared/src/components/notifications/NotificationItemAvatar.tsx]: Changed to barrel

File: packages/shared/src/components/modals/BootPopups.tsx (hunk 2 above)
Comment (omBratteng) [reply packages/shared/src/components/modals/BootPopups.tsx:251]: Most of it's now inside the `TopReaderBadgeModal` component, so we don't need to duplicate the logEvent calls. Then we can just set onAfterClose/ onAfterOpen props if we need anything extra.
I.e. in BootPopups we want to clear alerts after closing, and on notifications page want to revert the URL.

Comment (omBratteng) [reply packages/shared/src/components/modals/BootPopups.tsx:251]: Just makes it more dynamic, but keeps the common things in one place.

**Pull request merged**